packages/playground/src/hooks/useNavigationStatus.ts

```diff
@@ -0,0 +1,39 @@
+import noop from "lodash/fp/noop.js";
+import { onMounted, onUnmounted, ref } from "vue";
+import { useRouter } from "vue-router";
+
+export function useNavigationStatus() {
+  const router = useRouter();
+  const status = ref<"Loading" | "Success" | "Failed">("Success");
+
+  let unSubscribeBeforeEach = noop;
+  let unSubscribeAfterEach = noop;
+
+  let currentToPath: string;
+
+  onMounted(async () => {
+    await router.isReady();
+
+    unSubscribeBeforeEach = router.beforeEach((to, _, next) => {
+      currentToPath = to.fullPath;
+      status.value = "Loading";
+
+      return next();
+    });
+
+    unSubscribeAfterEach = router.afterEach((to, _, failure) => {
+      if (status.value !== "Loading" || currentToPath !== to.fullPath) {
+        return;
+      }
+
+      status.value = failure ? "Failed" : "Success";
```

Comment:
might be a good idea to display an error message if page failed to load. otherwise the progress bar just gets stuck like this 
![image](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/56790126/71489671-8d2b-4a0b-b818-83eb5fda19ff)

Reply:
well this is not how vue-router works ... vue-router will wait till the connection back and won't notify me that an error happen so I have no idea at that point if routing will fail or success

Reply:
Then what is the point of setting status to Failed if no action will be taken in that case

Reply:
router might fail to load component or something else